File: client/src/main/java/com/mthwate/conk/listener/PlayerPropertiesListener.java
package com.mthwate.conk.listener;

import com.jme3.network.Client;
import com.mthwate.conk.message.PlayerPropertiesMessage;
import com.mthwate.conk.state.MovementAppState;

/**
 * @author mthwate
 */
public class PlayerPropertiesListener extends AbstractClientListener<PlayerPropertiesMessage> {

	private final MovementAppState movementAppState;

	public PlayerPropertiesListener(MovementAppState movementAppState) {
		this.movementAppState = movementAppState;
	}

	@Override
	protected void onReceived(Client source, PlayerPropertiesMessage m) {
		movementAppState.initControl(m.getRadius(), m.getHeight(), m.getMass());
	}

}

File: client/src/main/kotlin/com/mthwate/conk/listener/PlayerPropertiesListener.kt
package com.mthwate.conk.listener

import com.jme3.network.Client
import com.mthwate.conk.message.PlayerPropertiesMessage
import com.mthwate.conk.state.MovementAppState

/**
 * @author mthwate
 */
class PlayerPropertiesListener(private val movementAppState: MovementAppState) : AbstractClientListener<PlayerPropertiesMessage>() {

	override fun onReceived(src: Client, msg: PlayerPropertiesMessage) {
		movementAppState.initControl(msg.radius, msg.height, msg.mass)
	}

}
